tap_mssql/sync_strategies/log_based.py

```diff
@@ -317,6 +317,20 @@ def sync_table(mssql_conn, config, catalog_entry, state, columns, stream_version
                         lsn_from,
                         lsn_to,
                     )
+                    if lsn_to == state_last_lsn:
+                        LOGGER.info(
+                            (
+                                "The last lsn processed as per the state file is equal to the max"
+                                " lsn available - no changes expected - state lsn will not be incremented"
+                            ),
+                        )
+                        from_lsn_expression = "{}".format(py_bin_to_mssql(state_last_lsn))
+                    else:
+                        from_lsn_expression = (
+                            (
+                                "sys.fn_cdc_increment_lsn({})"
+                            ).format(py_bin_to_mssql(state_last_lsn))
+                        )

```

Comment:
This fixes an issue where the LSN was previously incremented despite no records being extracted. This caused the tap to fail on a subsequent run as the logic assumed records had been missed because the LSN erroneously jumped ahead.